MHTimer.js

```diff
@@ -2458,42 +2525,131 @@ function updRH(message) {
+            }, (error, response, body) => {
+                if (!error && response.statusCode === 200 && body) {
+                    console.log(`Relic hunter location updated from ${relic_hunter.location} to ${body}`);
+                    resolve({
+                        source: 'DBGames',
+                        location: body,
+                        last_seen: 0
+                    });
+                } else {
+                    console.log(`RelicHunter query failed:`, error, response, body);
+                    resolve({
+                        source: 'broken',
+                        location: 'unknown',
+                        last_seen: 0
+                    });
+                }
+            });
+    });
+
+}
+
+/**
+ * Looks up the relic hunter location from MHCT
+ */
+async function MHCTRHLookup() {
+    if (relic_hunter.source === 'MHCT' && relic_hunter.location !== 'unknown') {
+        return;
+    }
+    return new Promise(function(resolve){
+        let req = request({
             uri: 'https://mhhunthelper.agiletravels.com/tracker.json',
             json: true
         }, (error, response, body) => {
             if (!error && response.statusCode === 200) {
-                if (body["rh"]["location"] === 'unknown') {
-                    if (relic_hunter.location)
-                        body["rh"]["location"] = relic_hunter.location;
-                }
-                responseStr = `Relic Hunter has been spotted in **${decode(body["rh"]["location"])}** and will be there for the next `;
-                responseStr += timeLeft(DateTime.utc().endOf('day'));
-                channel.send(responseStr);
-            }
-            else {
-                reportRequestError(`RelicHunter query failed:`, error, response, body);
-                channel.send("I was unable to get a good answer on that one.");
+                resolve({
+                        source: 'MHCT',
+                        location: decode(body["rh"]["location"]),
+                        last_seen: decode(body["rh"]["last_seen"])
+                    }
+                );
+            } else {
+                console.log(`RelicHunter query failed:`, error, response, body);
+                resolve({
+                    source: 'broken',
+                    location: 'unknown',
+                    last_seen: 0
+                    }
+                );
             }
         });
-    // If we have to use dbgames.info the xpath is: //*[@id="maincontent"]/table/tbody/tr/td[1]/div/div[4]/div[3]/h4/b/a
+    });
+}
+
+/**
+ * Processes a request to find the relic hunter
+ * @param {TextChannel} channel the channel on which to respond.
+ */
+async function findRH(channel) {
+    let responseStr = "";
+    let original_location = relic_hunter.location;
+    console.log(`Looking for RH, might be in ${original_location}`);
+    await getRHLocation();
```

Comment:
I think we should only invoke the lookup if the location is unknown.

Reply:
It is possible / likely that a sequence of events will cause DBGames not to know the new location (new hint added, for example). This has caused it to continue to report the old location. If MHCT has set the location (it automatically clears it every midnight) then someone has actually caught it there. So I do want to get the location unless it's !unknown && MHCT.